### ISSUE TITLE
pvs-studio analyze result

### ISSUE THREAD

**Issue by mcroitor** [thomas-maeder/popeye#216]
Verified code with pvs-studio. Found next errors:
- condition.c :1754 

```
  if (CondFlag[black_oscillatingKs] && OscillatingKings[White]==ConditionTypeC
      && CondFlag[white_oscillatingKs] && OscillatingKings[White]==ConditionTypeC)
    CondFlag[swappingkings] = true;
```

OscillatingKings[White]==ConditionTypeC repeats. Think first need to be with 'Black' index
- bouncy.c:157-170

```
 for (k= vec_knight_start; k <= vec_knight_end; k++)
    if (x) {
      sq_departure= i1+vec[k];
      if (is_square_empty(sq_departure)) {
        if (!NoEdge(sq_departure)) {
          if (rrefcech(sq_departure,x-1,evaluate))
            return true;
        }
      }
      else if (EVALUATE_OBSERVATION(evaluate,sq_departure,sq_target))
        return true;
    }
    else
      for (k= vec_knight_start; k <= vec_knight_end; k++) {
```

Is the normal for this part of code when cycle is declared in cycle and use the same counter with the same margins?

**Comment by thomas-maeder** [comment]
I agree on both accounts.

**Comment by thomas-maeder** [comment]
Fixed for 4.77
